testsuite/shared/src/main/java/org/jboss/as/test/module/util/TestModule.java

```diff
@@ -293,6 +318,25 @@ private File getModulesDirectory() {
         return moduleDir;
     }
 
+    private static boolean isImmutableModulePath(String path) {
+        return path.contains(ILLEGAL_BUILD_DIR) || path.contains(ILLEGAL_DIST_DIR);
+    }
+
+    private static boolean isUnderCurrentProjectBuildDir(File moduleRoot) {
+        String buildDir = System.getProperty("project.build.directory");
```

Comment:
CI errors are related; It looks like this will return always `null`, since `project.build.directory` is not passed by surefire maven plugin via system.properties. An alternative could be to use `Paths.get(System.getProperty("basedir"), "target").toString()`

${basedir} should not be null since it is always passed by surefire retrieved from the pom ${basedir} (although it is deprecated)

If so, I would bet this function will never return false

Reply:
Ups, I mean, never will return false due to the `if (buildDir != null)` check

Reply:
Thanks; I added that in; we'll see if it fixes it.

It's strange that the most recent run passed on Windows.